modules/learn/pages/clusters-and-availability/xdcr-with-scopes-and-collections.adoc

```diff
@@ -0,0 +1,167 @@
+A _namespace_ indicates the scope and collection, within the source or target bucket, within which exists data to be replicated.
+Each namespace is represented as the name of the scope, followed by the name of the collection, separated by a period.
+For example, if a bucket contains a scope named `HotelCollection`, and this scope contains collections named `USHotelCollection` and `UKHotelCollection`, then `HotelCollection.USHotelCollection` and `HotelCollection.UKHotelCollection` are both valid namespaces, specifying different collections within the same scope.
+
+Between a namespace on the source cluster and a namespace on the target cluster, two kinds of collections-mapping are supported, which are _implicit_ and _explicit_:
+
+* _Implicit_ mapping occurs whenever the source and target buckets contain an identical namespace: XDCR between each pair of identical namespaces is initiated automatically, as soon as replication starts between the source and target buckets.
+
+* _Explicit_ mapping is administrator-determined, and allows replication to occur between namespaces that are not identical.
+The correct pairings of namespaces on source and target must therefore be specified by the administrator.
+
+These forms of mapping are further described below.
+
+[#implicit-mapping]
+== Implicit Mapping
+
+Implicit mapping occurs whenever the source and target buckets contain an identical namespace: replication of documents between these namespaces occurs automatically, as soon as XDCR between the source and target buckets is commenced.
+This can be illustrated as follows:
+
+image::clusters-and-availability/xdcr-implicit-mapping-diagram.png[,540,align=left]
+
+The annotations are as follows:
+
+. The administrator must explicitly specify a target and a source bucket, between which replication is to occur.
+The buckets are typically on different clusters, in different data centers: however, they may also be on different clusters in the same data center, or even on the same cluster.
+
+. Every bucket contains a default scope and collection, whose namespace is `_default._default`.
+Implicit replication always occurs, therefore, between the default collection on the source bucket and the default collection on the target bucket.
+
+. In this example, the source bucket contains a scope named `ScopeA`, within which a collection named `CollectionA` exists.
+The target bucket also contains `CollectionA` within `ScopeA`.
+Therefore, since source and target buckets contain the identical namespace, `ScopeA.CollectionA`, implicit replication occurs automatically between `ScopeA.CollectionA` on the source, and `ScopeACollectionA` on the target.
++
+(Note that XDCR does not automatically create scopes or collections.
+Therefore, for an implicit mapping to be achieved, identical namespaces must have been previously created by the administrator.)
```

Comment:
For 1, I've changed to "identical namespaces must be created by the administrator".

Reply:
For 2, I've modified here to say "Note, however, if an identical namespace is _subsequently_ established within the target bucket, this is eventually detected by XDCR, by means of a periodic check." Then, there's a pointer to the section Target-Collection Removal and Addition. In that section, it now says: "XDCR checks the target namespaces every minute: when a check is performed, any new collections that have been added to the target are detected.
(Note, therefore, that it may indeed take XDCR up to 60 seconds to detect a newly created collection on the target: detection is _not_ instantaneous.)"

BTW, shall I assume that detecting collection removal is by the same process, and will therefore also take up to 60 seconds?

Reply:
> detecting collection removal

Just being explicitly clear, I think customers prob want to know about the XDCR behavior due to _target_ collection removal. (Source collection removal just means that no more data will be replicated since the source namespace is now gone). Anyway, the following is how the system is going to work in this scenario:

If a source collection has a mapped target collection, and there are ongoing mutations that are sent from the source collection, then XDCR will be able to detect the target collection removal as soon as it tries to send a mutation and finds out that the target collection is now gone. This detection mechanism does not depend on the per-minute interval and should be much quicker as it depends on the failure returned by the target cluster when sending a mutation.

If a mapped target collection is removed, but there is no mutation being sent to it, XDCR will detect the removal but will remain silent. 

(Essentially, it is replication-data driven)

Reply:
I've added a note on source-collection removal as follows:
If a source collection has a mapped target collection, and replication from the source collection is ongoing; if the source collection is removed, this is detected by XDCR as soon as XDCR tries to send a mutation. 

The detection of source-collection removal, therefore, does not depend on the 60-second interval required for the detection of target-collection removal; and is likely to occur much more quickly.